pennylane/measurements/measurements.py

```diff
@@ -693,3 +695,40 @@ def map_wires(self, wire_map: dict):
         else:
             new_measurement._wires = Wires([wire_map.get(wire, wire) for wire in self.wires])
         return new_measurement
+
+
+class SampleMeasurement(MeasurementProcess, ABC):
+    """Sample-based measurement process."""
+
+    @abstractmethod
+    def process(
+        self, samples: Sequence[complex], shot_range: Tuple[int] = None, bin_size: int = None
+    ):
+        """Process the given samples.
+
+        Args:
+            samples (Sequence[complex]): computational basis samples generated for all wires
+            shot_range (tuple[int]): 2-tuple of integers specifying the range of samples
+                to use. If not specified, all samples are used.
+            bin_size (int): Divides the shot range into bins of size ``bin_size``, and
+                returns the measurement statistic separately over each bin. If not
+                provided, the entire shot range is treated as a single bin.
+        """
+
+
+class StateMeasurement(MeasurementProcess, ABC):
+    """State-based measurement process."""
+
+    @abstractmethod
+    def process_state(self, state: np.ndarray, device_wires: Wires):
```

Comment:
Is there a reason this has a different name? Seems like it would make more sense for these names to be the same.

Reply:
Some measurements will inherit from both `SampleMeasurement` and `StateMeasurement` classes. Given that we cannot perform method overloading in python, we need to use different names.

Reply:
When supporting these methods in the device, we'll have something like the following:
```python
class Device:
    def statistics(self, measurements):
        results = []
        for m in measurements:
            if self.shots is None:
                results.append(m.process_state(self.state, self.wires)
            else:
                results.append(m.process(self._samples, shot_range, bin_size)
```